src/Modal.js

```diff
@@ -309,6 +338,7 @@ class Modal extends React.Component {
     }
 
     Modal.openCount += 1;
+    this.modalCount = Modal.openCount;
```

Comment:
Let's rename`this.modalCount` to `this.modalIndex` to be more clear.  Also move the assignment above `Modal.openCount += 1` to make it 0-based.